node-js-basics-101/node_modules/asynquence-contrib/contrib-common.src.js

```diff
@@ -0,0 +1,717 @@
+								// itself.
+								ret.value.length > 0 ? ret.value : ASQ.messages(undefined)
+							);
+						}
+						// non-undefined value returned?
+						else if (typeof ret.value !== "undefined") {
+							// wrap the value in a sequence
+							ret.value = ASQ(ret.value);
+						}
+						else {
+							// make an empty sequence
+							ret.value = ASQ();
+						}
+					}
+
+					ret.value
+					.val(function $$val(){
+						// bail on run in aborted sequence
+						if (internals("seq_aborted")) return;
+
+						if (arguments.length > 0) {
+							// save any return messages for input
+							// to next iteration
+							next_val = arguments.length > 1 ?
+								ASQ.messages.apply(ø,arguments) :
+								arguments[0]
+							;
+						}
+
+						// still more to iterate?
+						if (!ret.done) {
+							// was the control token passed along?
+							if (next_val === token) {
+								// round-robin: put co-routine back into the list
+								// at the end, so that the the next iterator can be
+								// processed on next loop-iteration
+								iterators.push(iter);
+							}
+							else {
+								// put co-routine back in where it just
+								// was so it can be processed again on
+								// next loop-iteration
+								iterators.unshift(iter);
+							}
+						}
+
+						iterateOrQuit(iterate,/*now=*/true);
+					})
+					.or(function $$or(){
+						// bail on run in aborted sequence
+						if (internals("seq_aborted")) return;
+
+						if (!ret.done) {
+							// put co-routine back in where it just
+							// was so it can be processed again on
+							// next loop-iteration
+							iterators.unshift(iter);
+
+							next_val = arguments.length > 1 ?
+								ASQ.messages.apply(ø,arguments) :
+								arguments[0];
+
+							iterate(/*throwNext=*/true);
+						}
+						else {
+							// if an error is left over but iteration now
+							// complete, pass out to the main sequence
+							mainDone.fail.apply(ø,arguments);
+						}
+					});
+				}
+			})();
+		});
+
+		return api;
+	};
+});
+// "toPromise"
+ASQ.extend("toPromise",function $$extend(api,internals){
+	return function $$to$promise() {
+		return new Promise(function $$executor(resolve,reject){
+			api
+			.val(function $$val(){
+				var args = ARRAY_SLICE.call(arguments);
+				resolve.call(ø,args.length > 1 ? args : args[0]);
+				return ASQ.messages.apply(ø,args);
+			})
+			.or(function $$or(){
+				var args = ARRAY_SLICE.call(arguments);
+				reject.call(ø,args.length > 1 ? args : args[0]);
+			});
+		});
+	};
+});
+// "wrap"
+"use strict";
+
+ASQ.wrap = function $$wrap(fn, opts) {
+	function checkThis(t, o) {
+		return !t || typeof window != "undefined" && t === window || typeof global != "undefined" && t === global ? o : t;
+	}
+
+	function paramSpread(gen) {
+		return regeneratorRuntime.mark(function paramSpread(token) {
+			return regeneratorRuntime.wrap(function paramSpread$(context$3$0) {
+				while (1) switch (context$3$0.prev = context$3$0.next) {
```

Comment:
__[jshint]__
* Expected a conditional expression and instead saw an assignment.

<!-- Comment by [SideCI](https://sideci.com) -->